client/service/src/client/client.rs

```diff
@@ -390,6 +425,26 @@ where
 			backend.commit_operation(op)?;
 		}
 
+		let (unpin_worker_sender, mut rx) =
+			tracing_unbounded::<Block::Hash>("unpin-worker-channel", 10_000);
+		let task_backend = Arc::downgrade(&backend);
+		spawn_handle.spawn(
+			"unpin-worker",
+			None,
+			async move {
+				loop {
+					if let Some(message) = rx.next().await {
+						if let Some(backend) = task_backend.upgrade() {
+							backend.unpin_block(&message);
+						} else {
+							log::warn!(target: "db", "Terminating unpin-worker, backend reference was dropped.");
+							return
+						}
+					}
+				}
```

Comment:
```suggestion
				while let Some(message) = rx.next().await {
					if let Some(backend) = task_backend.upgrade() {
						backend.unpin_block(&message);
					} else {
						log::warn!(target: "db", "Terminating unpin-worker, backend reference was dropped.");
						return;
					}
				}
				log::warn!(target: "db", "Terminating unpin-worker, stream terminated.");
			}
```
Probably we can return as soon as the stream is terminated?